modules/claims_api/app/sidekiq/claims_api/v2/disability_compensation_pdf_generator.rb

```diff
@@ -0,0 +1,115 @@
+# frozen_string_literal: true
+
+require 'claims_api/v2/disability_compensation_pdf_mapper'
+require 'pdf_generator_service/pdf_client'
+
+module ClaimsApi
+  module V2
+    class DisabilityCompensationPdfGenerator < DisabilityCompensationClaimServiceBase
+      EVSS_DOCUMENT_TYPE = 'L023'
+
+      def perform(claim_id, middle_initial) # rubocop:disable Metrics/MethodLength
+        log_job_progress('526 v2 PDF Generator job',
+                         claim_id,
+                         "526EZ PDF generator started for claim #{claim_id}")
+
+        auto_claim = get_claim(claim_id)
+
+        # Reset for a rerun on this
+        set_pending_state_on_claim(auto_claim) unless auto_claim.status == pending_state_value
+
+        pdf_data = get_pdf_data
+        mapped_claim = pdf_mapper_service(auto_claim.form_data, pdf_data, auto_claim.auth_headers,
+                                          middle_initial).map_claim
+        pdf_string = generate_526_pdf(mapped_claim)
```

Comment:
`get_pdf_data` returns an object with just the data field. in line 24, we again call `get_pdf_data` from inside `generate_526_pdf(mapped_claim)` so line 24 seems useless to me.

it's also being provided to the pdf_mapper_service and set to an instance variable... could we not just set a default value there.... and later we extract the data.attributes out from said object and place it inside the data. seems like there may be some unnecessary repetition.

Reply:
you are right it was unnecessary, removed that line